sound/soc/sof/sof-audio.c

```diff
@@ -443,7 +443,7 @@ sof_prepare_widgets_in_path(struct snd_sof_dev *sdev, struct snd_soc_dapm_widget
 	if (!widget_ops)
 		return 0;
 
-	if (!swidget || !widget_ops[widget->id].ipc_prepare || swidget->prepared)
+	if (!swidget || !widget_ops[widget->id].ipc_prepare)
```

Comment:
this is far from self-explanatory, you just add a patch dedicated to filter the multiple calls to prepare and now we remove the test here?

Not clear why we add the if (swidget->prepared) in some places and remove it where it used to be. This feels odd

Reply:
The comment still applies, what are we trying to do here?

Also it's not really 'repeated ipc_prepare', what if I understand well is that existing paths will be left alone, it's only the paths that fork that will be impacted.